GoogleSignIn/Sources/GIDGoogleUser.m

```diff
@@ -137,6 +135,36 @@ - (nullable NSString *)openIDRealm {
   return [_authState.lastTokenResponse.request.additionalParameters[kOpenIDRealmParameter] copy];
 }
 
+- (NSString *)lastTokenResponseToken {
+  return _authState.lastTokenResponse.idToken;
+}
+
+- (nullable NSDate *)idTokenExpirationDate {
+  return [[[OIDIDToken alloc] initWithIDTokenString:[self lastTokenResponseToken]] expiresAt];
+}
+
+- (NSString *)lastTokenResponseAccessToken {
+  return _authState.lastTokenResponse.accessToken;
+}
+
+- (NSDate *)accessTokenExpirationDate {
+  return _authState.lastTokenResponse.accessTokenExpirationDate;
+}
+
+- (void)updateTokens {
```

Comment:
The three new token properties are all derived from `_authState`.  Should we not use cached getters for them as we've done for `configuration`?

Reply:
After consideration I removed this method and updated tokens in the method `updateAuthState:profileData:`